File: src/main/java/edu/uitm/ev_reservation/entity/ChargingSession.java
package edu.uitm.ev_reservation.entity;

import java.util.Date;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "charging_sessions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ChargingSession {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  public Long id;

  @ManyToOne
  public User user;

  @ManyToOne
  public Vehicle vehicle;

  @ManyToOne
  public EVStation station;

  public int pumpNumber;

  public boolean isCompleted;

  public boolean isReserved;

  public Date startTime;
}

File: static/src/types/charging-session.ts
import type { EvStations } from './ev-stations'
import type { User } from './user'
import type { Vehicle } from './vehicle'

export interface ChargingSession {
  id: number
  user: User
  vehicle: Vehicle
  station: EvStations
  pumpNumber: number
  isCompleted: boolean
  isReserved: boolean
  isCharging: boolean
  startTime: string
}

export interface ChargingSessionPage {
  content: ChargingSession[]
  totalElements: number
  totalPages: number
  size: number
  number: number
  numberOfElements: number
  first: boolean
  last: boolean
}
